### PR TITLE
[feat] add issue and PR templates

### LINKED ISSUE #17: [feat] add Issue and PR templates

### PR DESCRIPTION
**Pull request by kr-anurag**
Closes #17

### CODE REVIEW

**Review by AnishDe12020 — CHANGES_REQUESTED** [review]
For the issue templates, replicate the same in [GitHub Issue Forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms). Also the Define you section is not needed. Instead, you can add a section with checkboxes to choose a contribution category (following the [all contributors specification](https://allcontributors.org/docs/en/emoji-key)). The Pull Request template will remain markdown as pull request forms aren't supported yet

**Comment by kr-anurag** [comment]
cool! okay

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        Thank you for taking the time to file a bug report. To address this bug as fast as possible, we need some information.
+  - type: input
+    id: browser
+    attributes:
+      label: Broswer
+      description: "Which Web Browser do you use? Please provide the version as well."
+      placeholder: "Google Chrome"
+    validations:
+      required: true  
+  - type: textarea
```
Comment (AnishDe12020) [inline .github/ISSUE_TEMPLATE/bug.yml:20]: Cna you remove this field and instead have 2 fields - current behaviour and expected behaviour

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    attributes:
+      label: Bug description
+      description: What happened?
+    validations:
+      required: true
+  - type: textarea
+    id: steps
+    attributes:
+      label: Steps to reproduce
+      description: Which steps do we need to take to reproduce this error?
```
Comment (AnishDe12020) [inline .github/ISSUE_TEMPLATE/bug.yml:31]: Add one more field called additional information with a description saying they can paste relevant screenshots, screen recording, logs or anything which might help give more context/information to the issue

File: .github/ISSUE_TEMPLATE/feature.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        ---
+        
+        Thank you for taking the time to suggest a new feature, we need some information.
+  - type: textarea
+    id: feature-description
+    attributes:
+      label: Feature description
+      description: Provide a synopsis about the feature requested
+    validations:
+      required: true
```
Comment (AnishDe12020) [inline .github/ISSUE_TEMPLATE/feature.yml:18]: add a dropdown for the type of feature requests. The options will be  - 
- New Tool
- New Feature in a Tool
- New Feature for the app as a whole

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!-- Replace <`issueNumber`> with the Issue number to link it with this PR -->
+<!-- Example: #1 links this PR to the first Issue-->
+resolves #`issueNumber`
+
+## Checklist
+
+- [ ] You've followed the code structure 
+- [ ] You've linked the PR to correct Issue
```
Comment (AnishDe12020) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: Also add passes the CI pipeline (lint, formatting, other tests).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+## Checklist
+
+- [ ] You've followed the code structure 
+- [ ] You've linked the PR to correct Issue
+
+## Proposed Changes
+
+  -
+  -
+  -
```
Comment (AnishDe12020) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Add a commented block which says that one can add checkboxes for proposed changes and update the message as they have committed the changes.

Comment (AnishDe12020) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: Also add one more field called additional information where one can add more info on the changes like screenshots, dependencies added/removed, major changes to UI/UX, any breaking changes

**Review by AnishDe12020 — CHANGES_REQUESTED** [review]
Please look at the comments I left and make the changes accordingly :D

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (kr-anurag) [reply .github/PULL_REQUEST_TEMPLATE.md:14]: cool! lemme see

**Comment by AnishDe12020** [comment]
Also please follow the [conventional commits convention](https://www.conventionalcommits.org/en/v1.0.0/) for commit messages

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -6,9 +6,14 @@ resolves #`issueNumber`
...
 
   -
   -
-  -
\ No newline at end of file
+  -
+
+## Additonal Info
+
+### You can attach screenshots here, updated dependencies (if any), any breaking changes, etc.
```
Comment (AnishDe12020) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: The description for additional info will be commented out, not a third level heading

**Review by AnishDe12020 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    attributes:
+      label: Current Behaviour
+      description: What happened?
+    validations:
+      required: true
+  - type: textarea
+    id: expected-behaviour
+    attributes:
+      label: Expected Behaviour
+      description: What should be the expected behaviour?
```
Comment (AnishDe12020) [inline .github/ISSUE_TEMPLATE/bug.yml:31]: this will be a required field

File: .github/ISSUE_TEMPLATE/bug.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  - type: textarea
```
Comment (AnishDe12020) [inline .github/ISSUE_TEMPLATE/bug.yml:32]: This will also be a required field

**Review by AnishDe12020 — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug.yml (hunk 8 above)
Comment (kr-anurag) [reply .github/ISSUE_TEMPLATE/bug.yml:34]: > This will also be a required field

hey 
i have updated the files as requested

**Review by AnishDe12020 — APPROVED** [review]

**Comment by AnishDe12020** [comment]
@all-contributors add @kr-anurag for project management

**Pull request merged**